arc_solvers/models/qa/README.md

```diff
@@ -0,0 +1,59 @@
+# Models for Question Answering
+This document contains description of the neural models for question answering.
+
+## Multi Choice QA
+
+### BiLSTM Max-out with Question to Choices Max Attention
+The [`QAMultiChoiceMaxAttention`](multi_choice/qa_multi_choice_max_att.py) model computes the attention interaction between question and choices context-encoded representations.
+
+A high-level description of the model is:
+1. Obtain a BiLSTM context representation of the token sequences of the
+`question` and each `choice`.
+2. Get an aggregated (single vector) representations for `question` and `choice` using element-wise `max` operation.
+3. Compute the attention score between `question` and `choice` as  `linear_layer([u, v, u - v, u * v])`, where `u` and `v` are the representations from Step 2.
+4. Select as answer the `choice` with the highest attention with the `question`.
```

Comment:
Attention seems odd here. It is FF layer that takes a vector as input and produces a scalar score. Generally attention takes two inputs and computes scores over individual elements of one of the inputs.  What do you think of just calling it a score i.e. 

3. Compute the score for the `choice` as  `linear_layer([u, v, u - v, u * v])`, where `u` and `v` are the `choice` and `question` representations respectively from Step 2.
4. Select as answer the `choice` with the highest score.

Reply:
Technically the attention function is attention(u,v) => { return linear_layer([u, v, u - v, u * v])}. I agree that score sounds better in the context. Fixed 3 and 4.